src/server.c:1020:12: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]

cowsql build failed on Fedora 44 with the error:

```bash
libtool: compile:  gcc -DPACKAGE_NAME=\"libcowsql\" -DPACKAGE_TARNAME=\"libcowsql\" -DPACKAGE_VERSION=\"1.15.9\" "-DPACKAGE_STRING=\"libcowsql 1.15.9\"" -DPACKAGE_BUGREPORT=\"https://github.com/cowsql/cowsql\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libcowsql\" -DVERSION=\"1.15.9\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DHAVE_WCHAR_H=1 -DSTDC_HEADERS=1 -D_ALL_SOURCE=1 -D_DARWIN_C_SOURCE=1 -D_GNU_SOURCE=1 -D_HPUX_ALT_XOPEN_SOCKET_API=1 -D_NETBSD_SOURCE=1 -D_OPENBSD_SOURCE=1 -D_POSIX_PTHREAD_SEMANTICS=1 -D__STDC_WANT_IEC_60559_ATTRIBS_EXT__=1 -D__STDC_WANT_IEC_60559_BFP_EXT__=1 -D__STDC_WANT_IEC_60559_DFP_EXT__=1 -D__STDC_WANT_IEC_60559_EXT__=1 -D__STDC_WANT_IEC_60559_FUNCS_EXT__=1 -D__STDC_WANT_IEC_60559_TYPES_EXT__=1 -D__STDC_WANT_LIB_EXT2__=1 -D__STDC_WANT_MATH_SPEC_FUNCS__=1 -D_TANDEM_SOURCE=1 -D__EXTENSIONS__=1 -DHAVE_PTHREAD_PRIO_INHERIT=1 -DHAVE_PTHREAD=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_ARPA_INET_H=1 -DHAVE_FCNTL_H=1 -DHAVE_STDINT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_SYS_SOCKET_H=1 -DHAVE_UNISTD_H=1 -D_FILE_OFFSET_BITS=64 -I. -std=c11 -g -fcf-protection --param=ssp-buffer-size=4 -pipe -fno-strict-aliasing -fdiagnostics-color -fexceptions -fstack-clash-protection -fstack-protector-strong -fasynchronous-unwind-tables -fdiagnostics-show-option -Wall -Wextra -Wimplicit-fallthrough=5 -Wcast-align -Wstrict-prototypes -Wlogical-op -Wmissing-include-dirs -Wold-style-definition -Winit-self -Wfloat-equal -Wsuggest-attribute=noreturn -Wformat=2 -Wshadow -Wendif-labels -Wdate-time -Wnested-externs -Wconversion -Werror -pthread -O2 -fvisibility=hidden -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m32 -march=i686 -mtune=generic -msse2 -mfpmath=sse -mstackrealign -fasynchronous-unwind-tables -fstack-clash-protection -mtls-dialect=gnu -c src/server.c  -fPIC -DPIC -o src/.libs/libcowsql_la-server.o
src/server.c: In function 'parseNodeStore':
src/server.c:1020:12: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
 1020 |         nl = memchr(p, '\n', (size_t)(end - version_str));
      |            ^
src/server.c:1032:20: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
 1032 |                 nl = memchr(p, '\n', (size_t)(end - addr));
      |                    ^
src/server.c:1040:20: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
 1040 |                 nl = memchr(p, '\n', (size_t)(end - id_str));
      |                    ^
src/server.c:1059:20: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
 1059 |                 nl = memchr(p, '\n', (size_t)(end - role_str));
      |                    ^
src/server.c: In function 'parseLocalInfo':
src/server.c:1156:12: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
 1156 |         nl = memchr(version_str, '\n', (size_t)(end - version_str));
      |            ^
src/server.c:1167:12: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
 1167 |         nl = memchr(addr, '\n', (size_t)(end - addr));
      |            ^
src/server.c:1175:12: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
 1175 |         nl = memchr(id_str, '\n', (size_t)(end - id_str));
      |            ^
cc1: all warnings being treated as errors
make: *** [Makefile:1596: src/libcowsql_la-server.lo] Error 1
make: *** Waiting for unfinished jobs....
```

Ref: https://kojipkgs.fedoraproject.org//work/tasks/8083/141148083/build.log
https://koji.fedoraproject.org/koji/taskinfo?taskID=141147089

--- a/src/server.c
+++ b/src/server.c
@@ -987,10 +987,10 @@ static const struct client_node_info *findNodeInCache(
  * representation. */
 static int parseNodeStore(char *buf, size_t len, struct node_store_cache *cache)
 {
-	const char *p = buf;
+	char *p = buf;
 	const char *end = buf + len;
 	char *nl;
-	const char *version_str;
+	char *version_str;
 	const char *addr;
 	const char *id_str;
 	const char *dig;
@@ -1126,12 +1126,12 @@ static int parseLocalInfo(char *buf,
 			  char **local_addr,
 			  uint64_t *local_id)
 {
-	const char *p = buf;
+	char *p = buf;
 	const char *end = buf + len;
 	char *nl;
-	const char *version_str;
-	const char *addr;
-	const char *id_str;
+	char *version_str;
+	char *addr;
+	char *id_str;
 	const char *dig;
 	unsigned long long id;
 

